test/boost/cql_query_test.cc

```diff
@@ -5064,6 +5064,104 @@ SEASTAR_TEST_CASE(test_parallelized_select_count) {
+        }
+        auto msg = e.execute_cql("SELECT MAX(k) FROM tbl;").get();
+        assert_that(msg).is_rows().with_rows({
+            {int32_type->decompose(int32_t(value_count - 1))}
+        });
+
+        BOOST_CHECK_EQUAL(stat_parallelized + 1, qp.get_cql_stats().select_parallelized);
+    });
+}
+
+SEASTAR_TEST_CASE(test_parallelized_select_sum) {
+    return do_with_cql_env_thread([](cql_test_env& e) {
+        auto& qp = e.local_qp();
+        auto stat_parallelized = qp.get_cql_stats().select_parallelized;
+
+        e.execute_cql("CREATE TABLE tbl (k int, PRIMARY KEY (k));").get();
+        int value_count = 10;
+        for (int i = 0; i < value_count; i++) {
+            e.execute_cql(format("INSERT INTO tbl (k) VALUES ({:d});", i)).get();
+        }
+        auto msg = e.execute_cql("SELECT SUM(k) FROM tbl;").get();
+        assert_that(msg).is_rows().with_rows({
+            {int32_type->decompose(int32_t((value_count - 1) * value_count / 2))}
+        });
+
+        BOOST_CHECK_EQUAL(stat_parallelized + 1, qp.get_cql_stats().select_parallelized);
+    });
+}
+
+SEASTAR_TEST_CASE(test_non_parallelized_multiple_select) {
+    return do_with_cql_env_thread([](cql_test_env& e) {
+        auto& qp = e.local_qp();
+        auto stat_parallelized = qp.get_cql_stats().select_parallelized;
+
+        e.execute_cql("CREATE TABLE tbl (k int, PRIMARY KEY (k));").get();
+        int value_count = 10;
+        for (int i = 0; i < value_count; i++) {
+            e.execute_cql(format("INSERT INTO tbl (k) VALUES ({:d});", i)).get();
+        }
+        auto msg = e.execute_cql("SELECT MIN(k), MAX(k) FROM tbl;").get();
+        assert_that(msg).is_rows().with_rows({
+            {int32_type->decompose(int32_t(0)), int32_type->decompose(int32_t(value_count - 1))}
+        });
+
+        BOOST_CHECK_EQUAL(stat_parallelized, qp.get_cql_stats().select_parallelized);
+    });
+}
+
+SEASTAR_TEST_CASE(test_parallelized_select_sum_group_by) {
+    return do_with_cql_env_thread([](cql_test_env& e) {
+        auto& qp = e.local_qp();
+        auto stat_parallelized = qp.get_cql_stats().select_parallelized;
+
+        e.execute_cql("CREATE TABLE tbl (k int, c int, v int, PRIMARY KEY (k, c));").get();
+        int value_count = 10;
+        for (int k = 0; k < 2; k++) {
+            for (int c = 0; c < value_count; c++) {
+                e.execute_cql(format("INSERT INTO tbl (k, c, v) VALUES ({:d}, {:d}, {:d});", k, c, c)).get();
+            }
+        }
+    
+        auto msg = e.execute_cql("SELECT k, SUM(v) FROM tbl GROUP BY k;").get();
+        assert_that(msg).is_rows().with_rows({
+            {int32_type->decompose(int32_t(1)), int32_type->decompose(int32_t((value_count - 1) * value_count / 2))},
+            {int32_type->decompose(int32_t(0)), int32_type->decompose(int32_t((value_count - 1) * value_count / 2))}
+        });
+
+        BOOST_CHECK_EQUAL(stat_parallelized, qp.get_cql_stats().select_parallelized);
+    });
+}
+
```

Comment:
tests for parallelized UDA?

Please add negative tests too for cases that aren't supported.

Reply:
tests added